plutus-core/plutus-core/src/PlutusCore/Constant/Dynamic/Emit.hs

```diff
@@ -34,6 +35,22 @@ instance Monad Emitter where
 instance MonadEmitter Emitter where
     emit str = Emitter $ emit str
 
+-- | A newtype wrapper for providing a 'MonadEmitter' instance by directly providing the function.
+newtype WithEmitterT m a = WithEmitterT
+    { unWithEmitterT :: (String -> m ()) -> m a
+    } deriving
+        ( Functor, Applicative, Monad
+        , MonadError e, MonadState s
+        )
+      via
+        ReaderT (String -> m ()) m
+
+instance Monad m => MonadEmitter (WithEmitterT m) where
+    emit s = WithEmitterT $ \e -> e s
+
+instance MonadTrans WithEmitterT where
+    lift a = WithEmitterT $ \_ -> a
```

Comment:
Should be via-derivable as well?

Reply:
It wouldn't let me for some reason, I think `MonadTrans` looks a bit too weird for it. Stuff like:

```
plutus-core/src/PlutusCore/Constant/Dynamic/Emit.hs:47:17: error:
    • Cannot eta-reduce to an instance of form
        instance (...) => MonadTrans WithEmitterT
    • In the newtype declaration for ‘WithEmitterT’
   |
47 |       deriving (MonadTrans) via ReaderT (String -> m ())
   |                 ^^^^^^^^^^

```

Reply:
Ah, I see, thanks. Probably worth a tiny comment.